pkg/worker/handler/operator/registry/error.go

```diff
@@ -0,0 +1,14 @@
+package registry
+
+import (
+	"errors"
+
+	"github.com/aws/aws-sdk-go-v2/service/ecr/types"
+)
+
+func isImageNotFound(err error) bool {
+	var inf *types.ImageNotFoundException
+	var rnf *types.RepositoryNotFoundException
+
+	return errors.As(err, &inf) || errors.As(err, &rnf)
+}
```

Comment:
The error handling for the AWS SDK is rather unfortunate, so we have to do some magic here in order to match those errors.